website_lazy_load_image/static/src/js/frontend.js

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2018 Onestein
+ * License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). */
+
+odoo.define('website_lazy_load_image.lazy_image_loader', function (require) {
+    'use strict';
+
+    var Class = require('web.Class');
+    var mixins = require('web.mixins');
+
+    var LazyImageLoader = Class.extend(mixins.EventDispatcherMixin, {
+        init: function() {
+            mixins.EventDispatcherMixin.init.call(this);
+            this.all_finished = $.Deferred();
+            this.plugin = $('#wrap img').data('loaded', false).lazy(
+                this.getPluginConfiguration()
+            );
+        },
+
+        getPluginConfiguration: function() {
+            return {
+                afterLoad: this._afterLoad.bind(this),
+                beforeLoad: this._beforeLoad.bind(this),
+                onError: this._onError.bind(this),
+                onFinishedAll: this._onFinishedAll.bind(this),
+                chainable: false
+            };
+        },
+
+        _beforeLoad: function(el) {
+            this.trigger('beforeLoad', el);
```

Comment:
are these events you trigger specific to the plugin? If yes I'd prefix them w/ 'lazyload_'

Reply:
Yes it's triggered by the lazyload plugin.